tests/integration/pilot/trafficshifting/helper.go

```diff
@@ -0,0 +1,102 @@
+// Copyright 2019 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package trafficshifting
+
+import (
+	"fmt"
+	"math"
+	"strings"
+	"sync"
+	"testing"
+	"time"
+
+	"istio.io/istio/pkg/test/framework/components/echo"
+)
+
+func sendTraffic(t *testing.T, duration time.Duration, batchSize int, from, to echo.Instance, hosts []string, weight []int32, errorBand float64) {
+	const totalThreads = 10
+	var hitCount sync.Map
+
+	for i := 0; i < totalThreads; i++ {
+		go func() {
+			resp, _ := from.Call(echo.CallOptions{
```

Comment:
I agree to add error handling. But I am not sure if it is good to fail the test when one of the requests failed because our target is calculating the distribution of large number of requests. For example, there are totally 200 requests and one of them failed. However, the other 199 requests obey the distribution, should we fail the test or pass?